src/3rd_party_remove.c

```diff
@@ -74,14 +98,29 @@ enum swupd_code third_party_remove_main(int argc, char **argv)
 
 	ret = swupd_init(SWUPD_NO_ROOT);
 	if (ret != SWUPD_OK) {
-		goto finish;
+		goto exit;
+	}
+
+	/* The last argument has to be the repo-name to be deleted */
+	err = third_party_remove_repo(argv[argc - 1]);
+	if (err < 0) {
+		if (err == -ENOENT) {
+			ret = SWUPD_INVALID_OPTION;
+		} else {
+			ret = SWUPD_COULDNT_WRITE_FILE;
+		}
+		goto exit;
+	}
+
+	if (remove_repo_directory(argv[argc - 1]) < 0) {
+		ret = SWUPD_COULDNT_REMOVE_FILE;
+		goto exit;
 	}
 
-	/* TODO Implement */
+	info("Repository %s and its contents removed successfully\n", argv[argc - 1]);
```

Comment:
I think this statement is grammatically incorrect,  the word "contents" in plural has a different meaning that what is intended here, I believe the right work is just "content" https://dictionary.cambridge.org/grammar/british-grammar/content-or-contents

Reply:
wow, I didn't know that. thanks